src/isolate.rs

```diff
@@ -66,6 +68,35 @@ impl IsolateState {
     let tx = maybe_tx.unwrap();
     tx.send((req_id, buf)).expect("tx.send error");
   }
+
+  fn update_metrics(&self, bytes_recv: u64, bytes_sent: u64) {
+    let mut g = self.metrics.lock().unwrap();
+    let metrics = g.deref_mut();
+
+    metrics.update(bytes_recv, bytes_sent);
+  }
+}
+
+pub struct Metrics {
```

Comment:
You might consider adding `[derive(Default)]` and then using `Default::default()` in `new()` (or even omit `new()` completely). I'm not sure if it's better.

Reply:
I used that, thanks!